### ISSUE TITLE
[naming-convention] "Enforce that private members are prefixed with an underscore" configuration is invalid

### ISSUE THREAD

**Issue by G-Rath** [typescript-eslint/typescript-eslint#1474]
This is using the example in the docs titled "Enforce that private members are prefixed with an underscore".

I've tried to narrow down what property is causing the issue, and it seems to be `modifier`, but I'm still upskilling on how to properly use `naming-convention` so can't say that for sure.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/naming-convention": [
      "error",
      {
        "selector": "memberLike",
        "modifier": ["private"],
        "format": ["camelCase"],
        "leadingUnderscore": "require"
      }
    ]
  }
}
```

```TS
class MyClass {
  private method() {
    //
  }
}
```

**Expected Result**
`method` is flagged by ESLint as requiring a leading underscore.

**Actual Result**
```
	Configuration for rule "@typescript-eslint/naming-convention" is invalid:
	Value {"selector":"memberLike","modifier":["private"],"format":["camelCase"],"leadingUnderscore":"require"} should NOT have additional properties.
	Value {"selector":"memberLike","modifier":["private"],"format":["camelCase"],"leadingUnderscore":"require"} should NOT have additional properties.
	Value {"selector":"memberLike","modifier":["private"],"format":["camelCase"],"leadingUnderscore":"require"} should NOT have additional properties.
	Value {"selector":"memberLike","modifier":["private"],"format":["camelCase"],"leadingUnderscore":"require"} should NOT have additional properties.
	Value {"selector":"memberLike","modifier":["private"],"format":["camelCase"],"leadingUnderscore":"require"} should NOT have additional properties.
	Value {"selector":"memberLike","modifier":["private"],"format":["camelCase"],"leadingUnderscore":"require"} should NOT have additional properties.
	Value {"selector":"memberLike","modifier":["private"],"format":["camelCase"],"leadingUnderscore":"require"} should NOT have additional properties.
	Value {"selector":"memberLike","modifier":["private"],"format":["camelCase"],"leadingUnderscore":"require"} should NOT have additional properties.
	Value {"selector":"memberLike","modifier":["private"],"format":["camelCase"],"leadingUnderscore":"require"} should NOT have additional properties.
	Value {"selector":"memberLike","modifier":["private"],"format":["camelCase"],"leadingUnderscore":"require"} should NOT have additional properties.
	Value {"selector":"memberLike","modifier":["private"],"format":["camelCase"],"leadingUnderscore":"require"} should NOT have additional properties.
	Value {"selector":"memberLike","modifier":["private"],"format":["camelCase"],"leadingUnderscore":"require"} should NOT have additional properties.
	Value {"selector":"memberLike","modifier":["private"],"format":["camelCase"],"leadingUnderscore":"require"} should NOT have additional properties.
	Value {"selector":"memberLike","modifier":["private"],"format":["camelCase"],"leadingUnderscore":"require"} should NOT have additional properties.
	Value {"selector":"memberLike","modifier":["private"],"format":["camelCase"],"leadingUnderscore":"require"} should NOT have additional properties.
	Value {"selector":"memberLike","modifier":["private"],"format":["camelCase"],"leadingUnderscore":"require"} should NOT have additional properties.
	Value {"selector":"memberLike","modifier":["private"],"format":["camelCase"],"leadingUnderscore":"require"} should NOT have additional properties.
	Value {"selector":"memberLike","modifier":["private"],"format":["camelCase"],"leadingUnderscore":"require"} should match exactly one schema in oneOf.
```

**Additional Info**

<details>
<summary>debug output</summary>

```
g-rath@DESKTOP-MUGH739:/c/Users/G-Rath/workspace/projects-ackama/eslint-config-ackama$ $(eslint . --debug)
  eslint:cli CLI args: [ '.', '--debug' ] +0ms
  eslint:cli Running on files +3ms
  eslint:config-array-factory Loading .eslintignore file: /c/Users/G-Rath/workspace/projects-ackama/eslint-config-ackama/.eslintignore +0ms
  eslint:ignore-pattern Create with: [ IgnorePattern { patterns: [ '/node_modules/*', '/bower_components/*' ], basePath: '/c/Users/G-Rath/workspace/projects-ackama/eslint-config-ackama', loose: false } ] +0ms
  eslint:ignore-pattern   processed: { basePath: '/c/Users/G-Rath/workspace/projects-ackama/eslint-config-ackama', patterns: [ '/node_modules/*', '/bower_components/*' ] } +33ms
  eslint:ignore-pattern Create with: [ IgnorePattern { patterns: [ '/node_modules/*', '/bower_components/*' ], basePath: '/c/Users/G-Rath/workspace/projects-ackama/eslint-config-ackama', loose: false } ] +29ms
  eslint:ignore-pattern   processed: { basePath: '/c/Users/G-Rath/workspace/projects-ackama/eslint-config-ackama', patterns: [ '/node_modules/*', '/bower_components/*' ] } +28ms
  eslint:file-enumerator Start to iterate files: [ '.' ] +0ms
  eslint:file-enumerator Directory: /c/Users/G-Rath/workspace/projects-ackama/eslint-config-ackama +29ms
  eslint:file-enumerator Enter the directory: /c/Users/G-Rath/workspace/projects-ackama/eslint-config-ackama +23ms
  eslint:cascading-config-array-factory Load config files for /c/Users/G-Rath/workspace/projects-ackama/eslint-config-ackama. +0ms
  eslint:cascading-config-array-factory No cache found: /c/Users/G-Rath/workspace/projects-ackama/eslint-config-ackama. +29ms
  eslint:config-array-factory Loading JS config file: /c/Users/G-Rath/workspace/projects-ackama/eslint-config-ackama/.eslintrc.js +282ms
  eslint:config-array-factory Config file found: /c/Users/G-Rath/workspace/projects-ackama/eslint-config-ackama/.eslintrc.js +28ms
  eslint:config-array-factory Loading parser "@typescript-eslint/parser" from /c/Users/G-Rath/workspace/projects-ackama/eslint-config-ackama/.eslintrc.js +30ms
  eslint:config-array-factory Loaded: @typescript-eslint/parser@2.16.1-alpha.14 (/c/Users/G-Rath/workspace/projects-ackama/eslint-config-ackama/node_modules/@typescript-eslint/parser/dist/parser.js) +31ms
  eslint:config-array-factory Loading plugin "@typescript-eslint/eslint-plugin" from /c/Users/G-Rath/workspace/projects-ackama/eslint-config-ackama/.eslintrc.js +220ms
  eslint:config-array-factory Loaded: @typescript-eslint/eslint-plugin@2.16.1-alpha.14 (/c/Users/G-Rath/workspace/projects-ackama/eslint-config-ackama/node_modules/@typescript-eslint/eslint-plugin/dist/index.js) +2ms
  eslint:config-array-factory Plugin /c/Users/G-Rath/workspace/projects-ackama/eslint-config-ackama/node_modules/@typescript-eslint/eslint-plugin/dist/index.js loaded in: 118ms +119ms
  eslint:cascading-config-array-factory No cache found: /c/Users/G-Rath/workspace/projects-ackama. +463ms
  eslint:config-array-factory Config file not found on /c/Users/G-Rath/workspace/projects-ackama +4ms
  eslint:cascading-config-array-factory No cache found: /c/Users/G-Rath/workspace. +30ms
  eslint:config-array-factory Config file not found on /c/Users/G-Rath/workspace +53ms
  eslint:cascading-config-array-factory No cache found: /c/Users/G-Rath. +55ms
  eslint:config-array-factory Config file not found on /c/Users/G-Rath +56ms
  eslint:cascading-config-array-factory No cache found: /c/Users. +55ms
  eslint:config-array-factory Config file not found on /c/Users +57ms
  eslint:cascading-config-array-factory No cache found: /c. +59ms
  eslint:config-array-factory Config file not found on /c +56ms
  eslint:cascading-config-array-factory No cache found: /. +59ms
  eslint:config-array-factory Config file not found on / +61ms
Error: .eslintrc.js:
        Configuration for rule "@typescript-eslint/naming-convention" is invalid:
        Value {"selector":"memberLike","modifier":["private"],"format":["camelCase"],"leadingUnderscore":"require"} should NOT have additional properties.
        Value {"selector":"memberLike","modifier":["private"],"format":["camelCase"],"leadingUnderscore":"require"} should NOT have additional properties.
        Value {"selector":"memberLike","modifier":["private"],"format":["camelCase"],"leadingUnderscore":"require"} should NOT have additional properties.
        Value {"selector":"memberLike","modifier":["private"],"format":["camelCase"],"leadingUnderscore":"require"} should NOT have additional properties.
        Value {"selector":"memberLike","modifier":["private"],"format":["camelCase"],"leadingUnderscore":"require"} should NOT have additional properties.
        Value {"selector":"memberLike","modifier":["private"],"format":["camelCase"],"leadingUnderscore":"require"} should NOT have additional properties.
        Value {"selector":"memberLike","modifier":["private"],"format":["camelCase"],"leadingUnderscore":"require"} should NOT have additional properties.
        Value {"selector":"memberLike","modifier":["private"],"format":["camelCase"],"leadingUnderscore":"require"} should NOT have additional properties.
        Value {"selector":"memberLike","modifier":["private"],"format":["camelCase"],"leadingUnderscore":"require"} should NOT have additional properties.
        Value {"selector":"memberLike","modifier":["private"],"format":["camelCase"],"leadingUnderscore":"require"} should NOT have additional properties.
        Value {"selector":"memberLike","modifier":["private"],"format":["camelCase"],"leadingUnderscore":"require"} should NOT have additional properties.
        Value {"selector":"memberLike","modifier":["private"],"format":["camelCase"],"leadingUnderscore":"require"} should NOT have additional properties.
        Value {"selector":"memberLike","modifier":["private"],"format":["camelCase"],"leadingUnderscore":"require"} should NOT have additional properties.
        Value {"selector":"memberLike","modifier":["private"],"format":["camelCase"],"leadingUnderscore":"require"} should NOT have additional properties.
        Value {"selector":"memberLike","modifier":["private"],"format":["camelCase"],"leadingUnderscore":"require"} should NOT have additional properties.
        Value {"selector":"memberLike","modifier":["private"],"format":["camelCase"],"leadingUnderscore":"require"} should NOT have additional properties.
        Value {"selector":"memberLike","modifier":["private"],"format":["camelCase"],"leadingUnderscore":"require"} should NOT have additional properties.
        Value {"selector":"memberLike","modifier":["private"],"format":["camelCase"],"leadingUnderscore":"require"} should match exactly one schema in oneOf.

    at validateRuleOptions (/c/Users/G-Rath/workspace/projects-ackama/eslint-config-ackama/node_modules/eslint/lib/shared/config-validator.js:138:19)
    at /c/Users/G-Rath/workspace/projects-ackama/eslint-config-ackama/node_modules/eslint/lib/shared/config-validator.js:193:9
    at Array.forEach (<anonymous>)
    at validateRules (/c/Users/G-Rath/workspace/projects-ackama/eslint-config-ackama/node_modules/eslint/lib/shared/config-validator.js:190:30)
    at validateConfigArray (/c/Users/G-Rath/workspace/projects-ackama/eslint-config-ackama/node_modules/eslint/lib/shared/config-validator.js:337:9)
    at CascadingConfigArrayFactory._finalizeConfigArray (/c/Users/G-Rath/workspace/projects-ackama/eslint-config-ackama/node_modules/eslint/lib/cli-engine/cascading-config-array-factory.js:417:13)
    at CascadingConfigArrayFactory.getConfigArrayForFile (/c/Users/G-Rath/workspace/projects-ackama/eslint-config-ackama/node_modules/eslint/lib/cli-engine/cascading-config-array-factory.js:271:21)
    at FileEnumerator._iterateFilesRecursive (/c/Users/G-Rath/workspace/projects-ackama/eslint-config-ackama/node_modules/eslint/lib/cli-engine/file-enumerator.js:396:49)
    at _iterateFilesRecursive.next (<anonymous>)
    at FileEnumerator.iterateFiles (/c/Users/G-Rath/workspace/projects-ackama/eslint-config-ackama/node_modules/eslint/lib/cli-engine/file-enumerator.js:251:49)
```
</details>

**Versions**

Also tried against canary.

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.16.0` |
| `@typescript-eslint/parser`        | `2.16.0` |
| `TypeScript`                       | `3.7.5` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `12.14.1` |
| `npm`                              | `6.13.6` |
